### PR TITLE
When a user has no roles, logout on button click

### PR DESCRIPTION
**Pull request by leanneeliatra**
### Description

Cypress tests for testing the no role button. On click, the user should be logged out.
https://github.com/opensearch-project/security-dashboards-plugin/pull/1564 

### Issues Resolved

https://github.com/opensearch-project/security-dashboards-plugin/pull/1564

### Check List

- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kavilla** [comment]
@leanneeliatra looks like changes were requested on the original PR, should that be addressed here too?

**Review by derek-ho — APPROVED** [review]
May need to change the button text here as well if that is done

File: cypress/integration/plugins/security-dashboards-plugin/missing_role_button_logout.js
Comment (ruanyl) [inline cypress/integration/plugins/security-dashboards-plugin/missing_role_button_logout.js:1]: I guess we use 2 spaces indentation, while this file looks like 4 spaces

File: cypress/integration/plugins/security-dashboards-plugin/missing_role_button_logout.js
Comment (ruanyl) [inline cypress/integration/plugins/security-dashboards-plugin/missing_role_button_logout.js:5]:
```suggestion
if (Cypress.env('SECURITY_ENABLED')) {
```
I wonder if you have prettier editor plugin? It helps to auto-format the code if you want.

**Comment by leanneeliatra** [comment]
@ruanyl @wanglam @Hailong-am @raintygao could the work in this MR please be reviewed, I have been in discussions to get the bug change itself over the line and when that is ready, I don't want the test to fall behind! If it could be reviewed and ready that would be great, thank you.

**Review by ruanyl — APPROVED** [review]

**Comment by Hailong-am** [comment]
@leanneeliatra would you mind fix the lint check https://github.com/opensearch-project/opensearch-dashboards-functional-test/actions/runs/6470978871/job/17568385980?pr=830

**Comment by leanneeliatra** [comment]
> @leanneeliatra would you mind fix the lint check https://github.com/opensearch-project/opensearch-dashboards-functional-test/actions/runs/6470978871/job/17568385980?pr=830

I was having issues doing this change locally yesterday, then I was getting 347 linting errors on other files. I made the change by editing the file here, in GitHub. If you have any advice about this please share.

Thanks @Hailong-am. The lint fix is now complete.

**Comment by leanneeliatra** [comment]
I'm not sure if I should be requesting reviews in the meantime, if all the checks aren't passed but in case it is okay to proceed (let me know if I should wait) can I please get a review @wanglam, @raintygao, @Hailong-am

**Comment by kavilla** [comment]
@leanneeliatra ive been pulling stuff down and verifying against a targetted release candidate as i still need to fix the ci here.

In the meantime can you post screen shots of these test passing and what is ran against? Also could post a screen shot of the dashboard sanity tests result as it will help verify this PR doesnt interfer with other tests.

This will help other maintainers with velocity

**Comment by leanneeliatra** [comment]
> In the meantime can you post screen shots of these test passing and what is ran against? Also could post a screen shot of the dashboard sanity tests result as it will help verify this PR doesnt interfer with other tests.

For sure thanks @kavilla and thanks very much for checking this for me too. 
I will add the test images.
For this request:  _Also could post a screen shot of the dashboard sanity tests result as it will help verify this PR doesn't interfere with other tests._ I'm not certain what tests you are referring to here/where to run them please.

**Comment by leanneeliatra** [comment]
@kavilla I ran the test locally & am attaching screenshots:
Testing the changes and tests locally.
repo: [opensearch-dashboards-functional-test](https://github.com/opensearch-project/opensearch-dashboards-functional-test) 
branch: leanneeliatra:missing-roles-button-tests
repo: [security-dashboards-plugin](https://github.com/opensearch-project/security-dashboards-plugin)
branch: leanneeliatra:2.8

**Comment by leanneeliatra** [comment]
Are there remaining tests needed here from my end? I've added the passing cypress tests locally - perhaps this is enough. 

Can you confirm please @RyanL1997, are the screenshots of the runs above sufficient or did you get to look at the cypress pipeline for this issue please? If so, could you add the final review and a comment please so all checks are passing (except cypress)? Thanks a million for all your help.

**Comment by leanneeliatra** [comment]
The fix itself has now been merged https://github.com/opensearch-project/security-dashboards-plugin/pull/1564
If the associated tests (this MR) could be reviewed and merged that would be great. The faling cypress test check above is due to cypress itself, the successfully running local test can be seen in a screenshot above.

Can this be merged please? @ruanyl could you confirm that for me please? I seem to have 2 reviews, yourself and @derek-ho, but I still need one more as only one of these is registering. Is that correct? Thanks.

**Comment by ruanyl** [comment]
@leanneeliatra I cannot perform merge right now, still need one more approval.

**Comment by leanneeliatra** [comment]
Thank you for checking this @ruanyl , I am requesting the last review via slack and PR. 

The failing cypress test check above is due to cypress itself, the successfully running local test can be seen in a screenshot above.

The associated bug fix is merged, can I please get a final review of the tests (this PR).  Thanks very much. cc @Hailong-am , @xluo-aws , @SuZhou-Joe

**Review by Hailong-am — APPROVED** [review]

**Pull request merged**
